js_modules/dagster-ui/packages/ui-core/src/asset-data/AssetLiveDataProvider.tsx

```diff
@@ -143,33 +157,55 @@ export const AssetLiveDataProvider = ({children}: {children: React.ReactNode}) =
 
   const isDocumentVisible = useDocumentVisibility();
 
+  const [isGloballyRefreshing, setIsGloballyRefreshing] = React.useState(false);
+  const [oldestDataTimestamp, setOldestDataTimestamp] = React.useState(0);
+
+  const onUpdatingOrUpdated = React.useCallback(() => {
+    const allAssetKeys = Object.keys(_assetKeyListeners).filter(
+      (key) => _assetKeyListeners[key]?.length,
+    );
+    let isRefreshing = allAssetKeys.length ? true : false;
+    let oldestDataTimestamp = Infinity;
+    for (const key of allAssetKeys) {
+      if (lastFetchedOrRequested[key]?.fetched) {
+        isRefreshing = false;
+      }
```

Comment:
It seems like this "global refreshing" state goes to TRUE when you click the button and then FALSE as soon as /any/ asset exists in the cache again? I feel like it'd be nice if the spinner continued until ALL of the assets onscreen had been refetched. 

Could we instead express this as `isRefreshing = allAssetKeys.every(key => lastFetchedOrRequested[key]?.fetched)`? 

This is another piece of state that feels derived from cache and maybe doesn't need to be separate. Maybe one thing that would make this simpler / easier to express would be copying the fetched times into the `cache` along with the result data. It feels like a lot of this is being written as imperative state changes because you can't use `lastFetchedOrRequested` as a dependency, but maybe copying the timestamps from there into `cache` when you copy the data would fix that?

Reply:
Most importantly It depends on `_assetKeyListeners` which I wanted to keep out of React state to avoid re-renders. If I moved this out of global state then I'd need to move all of the other functions relying on it into the main Provider instead. I could do that but then the provider becomes really big and I think it makes the file a bit harder to understand